docs/ilm.asciidoc

```diff
@@ -0,0 +1,190 @@
+ILM clears all index settings and sets up a predefined set of indices and policies that the user cannot initially configure.
+
+It is recommended to setup index lifecycle management (ILM) before starting APM Server.
+This prevents indices from being created before policies are changed, and allows policies to be applied to indices at creation time.
+It also excludes setup from the ingestion process, which allows you to ensure ILM is setup correctly before using APM.
+
+Index lifecycle management will manage an index based on its defined policy.
+Run the <<setup-command>> with the ` --index-management` flag to set up the default ILM policy:
+
+[source,js]
+-----------------------
+./apm-server setup --index-management
+-----------------------
+// CONSOLE
+
+//Policies only need to be created once, and will persist through version upgrades.
+You can confirm the policy was created with the GET lifecycle policy API.
+Here's what the transaction response looks like:
+
+[source,js]
+-----------------------
+GET _ilm/policy
+{
+    "apm-7.2.0-transaction": {
+        "version": 1,
+        "modified_date": "2019-05-28T15:55:26.791Z",
+        "policy": {
+            "phases": {
+                "warm": {
+                    "min_age": "31d",<3>
+                    "actions": {
+                        "readonly": {},
+                        "set_priority": {
+                            "priority": 50<4>
+                        }
+                    }
+                },
+                "hot": {
+                    "min_age": "0ms",
+                    "actions": {
+                        "rollover": {
+                            "max_size": "50gb",<1>
+                            "max_age": "7d"<2>
+                        },
+                        "set_priority": {
+                            "priority": 100<4>
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+-----------------------
+// CONSOLE
+<1> Rollover after _50gb_
+<2> Rollover after _7 days_
+<3> Move to warm phase after _31 days_
+<4> Priority for recovering your indices after a node restart. Higher priorities are recovered first.
+
+[float]
+[[ilm-edit]]
+=== Editing ILM defaults
+
+Once you've enabled index lifecycle management and setup the ILM policies, you can view and edit the policies in the Index lifecycle policies UI in Kibana. For more information about working with the UI,
+see {kibana-ref}/index-lifecycle-policies.html[*Index lifecycle policies*].
+
+Index policies and templates are prefixed with version numbers, for example: `apm-7.2.0-metric-000001`.
+This means if you use APM Server, modify the policy, and upgrade, you'll need to do two things:
+1. Set `ilm.enabled` to false and configure your indices to the old ILM indices in the yml file.
+2. Update the policies manually when you upgrade. When APM Server is started for the first time,
```

Comment:
No check for policies when `ilm.enabled` is set to false. There are additional `template`s created per event type holding minimal ILM information, if ilm is disabled in the server config, but should be used with a manual setup, the user has to take care of creating those templates upfront and setting `setup.template.overwrite: false` (default).

Reply:
I'm wondering if we should include this information (and some of the info in https://github.com/elastic/apm-server/issues/2253) in the troubleshooting section instead? I don't want to muddy the water too much, but if you think there will be a lot of users enabling/disabling/enabling/disabling, then we probably should include it here.